aicssegmentation/structure_wrapper/WorkflowStep.py

```diff
@@ -66,7 +72,30 @@ def execute(self, image: List[np.ndarray]) -> np.ndarray:
             try:
                 # Most functions require unpacking the images
                 self.result: np.ndarray = self.__function(*image)
-            except KeyError:
+            except (AttributeError, TypeError):
                 # Some functions want it as a list
                 self.result: np.ndarray = self.__function(image)
         return self.result
+
+    def get_params(self) -> Dict[str, Any]:
+        """
+        Get the parameter names and its default values for this step.
+
+        Params:
+            none
+
+        Returns:
+            (Dict[str, Any]): map of parameter names to default values. Default values
```

Comment:
So we'll have access to the parameter name and the default value, but not the parameter type. Are you planning on inferring the type from the value when we generate the widgets? Will there be a potential issue differentiating a float from an int?

Reply:
There's multiple types the value of this dict could be. Default values come as lists floats and ints in jianxu's json. We can still get the type we need this parameter to be from all_functions.json. Looking into a clean solution for this in my widget generation ticket.